src/main/java/com/slack/kaldb/server/KaldbLocalSearcher.java

```diff
@@ -38,33 +39,41 @@ public static SearchQuery fromSearchRequest(KaldbSearch.SearchRequest searchRequ
   }
 
   public static SearchResult<LogMessage> fromSearchResultProto(
-      KaldbSearch.SearchResult protoSearchResult) throws IOException {
-    List<LogMessage> hits = new ArrayList<>(protoSearchResult.getHitsCount());
-
-    for (ByteString bytes : protoSearchResult.getHitsList().asByteStringList()) {
-      LogWireMessage hit = JsonUtil.read(bytes.toStringUtf8(), LogWireMessage.class);
-      LogMessage message = LogMessage.fromWireMessage(hit);
-      hits.add(message);
-    }
-    List<HistogramBucket> histogramBuckets = new ArrayList<>();
-    for (KaldbSearch.HistogramBucket protoBucket : protoSearchResult.getBucketsList()) {
-      histogramBuckets.add(new HistogramBucket(protoBucket.getLow(), protoBucket.getHigh()));
+      KaldbSearch.SearchResult protoSearchResult) {
+    try {
+      List<LogMessage> hits = new ArrayList<>(protoSearchResult.getHitsCount());
+
+      for (ByteString bytes : protoSearchResult.getHitsList().asByteStringList()) {
+        LogWireMessage hit = JsonUtil.read(bytes.toStringUtf8(), LogWireMessage.class);
+        LogMessage message = LogMessage.fromWireMessage(hit);
+        hits.add(message);
+      }
+      List<HistogramBucket> histogramBuckets = new ArrayList<>();
+      for (KaldbSearch.HistogramBucket protoBucket : protoSearchResult.getBucketsList()) {
+        histogramBuckets.add(new HistogramBucket(protoBucket.getLow(), protoBucket.getHigh()));
+      }
+
+      return new SearchResult<>(
+          hits,
+          protoSearchResult.getTookMicros(),
+          protoSearchResult.getTotalCount(),
+          histogramBuckets,
+          protoSearchResult.getFailedNodes(),
+          protoSearchResult.getTotalNodes(),
+          protoSearchResult.getTotalSnapshots(),
+          protoSearchResult.getSnapshotsWithReplicas());
+    } catch (IOException e) {
+      // TODO
+      return new SearchResult<>(null, 1, 1, null, 0, 0, 0, 0);
```

Comment:
Why don't we return null in this case? If I cannot successfully parse a protomessage I would expect either an exception, or a null to be returned. Or we could make this return an `Optional` of a search result.